### DISCUSSION TITLE
Body is empty on post request to custom endpoint

### DISCUSSION THREAD

**Discussion by NickoftheSouth** [directus/directus#16036] [Archive]
I'm developing an endpoint extension for my Directus instance which requires a lot of data that I'd like to pass through as a body.

But all I'm getting in the request body is an empty set `{}`

Here's a minimal code example of my `endpoint\index.ts`

```js
import { defineEndpoint } from '@directus/extensions-sdk';
export default defineEndpoint((router) => {//{ services, _exceptions }
	const express = require('express');
	router.use(express.json()),
	router.post('/', (_req, res) => {
		console.log(req.body);
		res.send(JSON.stringify(req.body));
	}),
});
```
Which I believe should echo out the passed in body, but it's always just `{}`

What am I missing here?

**Comment by azrikahar** [comment]
You should be able to remove the require statement as it's an existing middleware, so it can be simplified as:

```js
import { defineEndpoint } from '@directus/extensions-sdk';

export default defineEndpoint((router) => {
	router.post('/', (req, res) => {
		console.log(req.body);
		res.send(JSON.stringify(req.body)); // alternatively you can use res.json(req.body)
	});
};
``` 

> Which I believe should echo out the passed in body, but it's always just `{}`

Can you try making sure `Content-Type: application/json` header is sent? `express.json()` looks for that by default, or we will only get an empty object otherwise. [Reference in Express docs](https://expressjs.com/en/api.html#express.json)

Quote from the docs: A new body object containing the parsed data is populated on the request object after the middleware (i.e. `req.body`), or an empty object (`{}`) if there was no body to parse

|Without the correct request header|With the correct request header|
|---|---|
|![](https://user-images.githubusercontent.com/42867097/196333146-27bae4be-4351-4897-9d5e-1b079a3ba9ca.png)|![](https://user-images.githubusercontent.com/42867097/196333184-5957a915-b265-4448-a850-ecfcbbcda099.png)|

**Reply by NickoftheSouth** [reply]
Yep, that was exactly the issue, the `Content-Type` header was missing. 
thank you!
